src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -714,6 +714,60 @@ impl Tree {
         node.children.is_empty() && !live.contains(&node.tag)
     }
 
+    fn can_be_replaced_by_single_child(
+        &self,
+        idx: UniIndex,
+        live: &FxHashSet<BorTag>,
+        global: &GlobalState,
+    ) -> Option<UniIndex> {
+        let node = self.nodes.get(idx).unwrap();
+        if node.children.len() != 1
+            || live.contains(&node.tag)
+            || node.parent.is_none()
+            || global.borrow().protected_tags.contains_key(&node.tag)
+        {
+            return None;
+        }
+        let child_idx = node.children[0];
+        let child = self.nodes.get(child_idx).unwrap();
+        if global.borrow().protected_tags.contains_key(&child.tag) {
+            // todo think really hard whether we can't just handle this as we would regularly
+            return None;
+        }
```

Comment:
I am not sure if this check is necessary. I reckon that it is not, i.e. we can merge protected tags up. Part of the intuition is:
* If the child is initialized at an offset, so was the parent at that offset.
* Almost all the protector does is make more transitions UB. Since we do not remove the protector, these same transitions remain UB
* The one exception is that `Reserved` becomes conflicted on foreign accesses -- but this should not cause any issues, since the foreign read still happens similarly on the child.